ichnaea/webapp/app.py

```diff
@@ -38,7 +38,7 @@ def wsgi_app(environ, start_response):
     global _APP
 
     if _APP is None:
-        _APP = main(ping_connections=True)
+        _APP = log_middleware(main(ping_connections=True))
```

Comment:
`main` returns a WSGI app function. All `wsgi_app` does is cache that app so it doesn't keep getting recreated. I think you can replace the `serve` line below with this:

```
        serve(
            wsgi_app,
            host="0.0.0.0",
            port=8000,
            expose_tracebacks=settings("local_dev_env"),
        )
```

I think that reduces some of the duplicity and confusion.

Reply:
True. I'll try that tomorrow.